pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -134,14 +123,26 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	if err := c.client.List(ctx, list, client.InNamespace(wl.Namespace), client.MatchingFields{RequestsOwnedByWorkloadKey: wl.Name}); client.IgnoreNotFound(err) != nil {
 		return reconcile.Result{}, err
 	}
+
 	ownedPrs := list.Items
 	activeOrLastPRForChecks := c.activeOrLastPRForChecks(ctx, wl, relevantChecks, ownedPrs)
 
+	err = c.syncCheckStates(ctx, wl, relevantChecks, activeOrLastPRForChecks)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !workload.HasQuotaReservation(wl) || workload.IsFinished(wl) {
+		// 1.2 workload has no reservation or is finished
+		log.V(5).Info("workload with no reservation or is finished")
```

Comment:
since we are not doing anything after this, I would also remove the log.

Reply:
Done